plugin-infra/plugin-metadata-store/src/com/thoughtworks/go/plugin/access/analytics/AnalyticsMetadataStore.java

```diff
@@ -34,4 +34,10 @@ public static AnalyticsMetadataStore instance() {
         return store;
     }
 
+    public void updateAssetsPath(String pluginId, String assetPath) {
+        AnalyticsPluginInfo pluginInfo = getPluginInfo(pluginId);
+        if (pluginInfo != null) {
```

Comment:
Isn't it fatal if pluginInfo is null? There's only one chance to set static assets. I'd think we would rather not silently fail.

Reply:
Well theoretically this should never happen, but you are right we as well not silently fail if this happens.